tsl/test/expected/hypercore_ddl.out

```diff
@@ -53,10 +48,144 @@ select chunk, amname from chunk_info where hypertable = 'readings'::regclass;
+             compress_chunk              
+-----------------------------------------
+ _timescaledb_internal._hyper_1_1_chunk
+ _timescaledb_internal._hyper_1_2_chunk
+ _timescaledb_internal._hyper_1_3_chunk
+ _timescaledb_internal._hyper_1_4_chunk
+ _timescaledb_internal._hyper_1_10_chunk
+(5 rows)
+
+select count(*) from readings where location is null;
+ count 
+-------
+     1
+(1 row)
+
+-- Pick a chunk to play with that is not the first chunk. This is
 -- mostly a precaution to make sure that there is no bias towards the
 -- first chunk and we could just as well pick the first chunk.
 select chunk from show_chunks('readings') x(chunk) limit 1 offset 3 \gset
+-- We should not be able to set the not null before we have removed
+-- the null rows in the table. This works for hypercore-compressed
+-- chunks but not for heap-compressed chunks.
+\set ON_ERROR_STOP 0
+alter table readings alter column location set not null;
+ERROR:  column "location" of relation "_hyper_1_10_chunk" contains null values
+\set ON_ERROR_STOP 1
+delete from readings where location is null;
+select count(*) from readings where location is null;
+ count 
+-------
+     0
+(1 row)
+
+\d readings
+                       Table "public.readings"
+  Column  |           Type           | Collation | Nullable | Default 
+----------+--------------------------+-----------+----------+---------
+ time     | timestamp with time zone |           | not null | 
+ location | integer                  |           |          | 
+ device   | integer                  |           | not null | 
+ temp     | numeric(4,1)             |           |          | 
+ humidity | double precision         |           |          | 
+ jdata    | jsonb                    |           |          | 
+Indexes:
+    "readings_time_key" UNIQUE CONSTRAINT, btree ("time")
+Triggers:
+    ts_insert_blocker BEFORE INSERT ON readings FOR EACH ROW EXECUTE FUNCTION _timescaledb_functions.insert_blocker()
+Number of child tables: 5 (Use \d+ to list them.)
+
+\d :chunk
+            Table "_timescaledb_internal._hyper_1_4_chunk"
+  Column  |           Type           | Collation | Nullable | Default 
+----------+--------------------------+-----------+----------+---------
+ time     | timestamp with time zone |           | not null | 
+ location | integer                  |           |          | 
+ device   | integer                  |           | not null | 
+ temp     | numeric(4,1)             |           |          | 
+ humidity | double precision         |           |          | 
+ jdata    | jsonb                    |           |          | 
+Indexes:
+    "4_4_readings_time_key" UNIQUE CONSTRAINT, btree ("time")
+Check constraints:
+    "constraint_4" CHECK ("time" >= 'Fri Jun 03 17:00:00 2022 PDT'::timestamp with time zone AND "time" < 'Sat Jun 04 17:00:00 2022 PDT'::timestamp with time zone)
+Inherits: readings
+
+alter table readings alter column location set not null;
```

Comment:
Note that after having deleted all rows that have null values, the chunks might not be compressed anymore or only partially compressed. We might want to test that we can set not null when we know there is compressed data.

Either recompress everything again or check that there is still compressed data.

Reply:
Compressed all the data.